src/main/java/domain/card/Deck.java

```diff
@@ -0,0 +1,17 @@
+package domain.card;
+
+import java.util.Collections;
+import java.util.List;
+
+public class Deck {
+    private final List<Card> cards;
+
+    public Deck() {
+        cards = CardFactory.create();
+        Collections.shuffle(cards);
+    }
+
+    public Card deal() {
+        return cards.remove(0);
```

Comment:
`CardFactory`를 보니 자료구조가 ArrayList네요.
0번째 인덱스 값을 지운다면 `deal` 메서드가 O(N)의 시간복잡도를 가지게 될 것 같은데, 아래와 같이 바꿔보는건 어떨까요?

```suggestion
        return cards.remove(cards.size() - 1);
```